src/Security/JsonWebTokenAuthenticator.php

```diff
@@ -33,126 +33,68 @@ public function __construct(protected JsonWebTokenManager $jwtManager, protected
-            'X-JWT-Authorization header required with JWT token. See ' .
-            "<a href='${apiDocsUrl}'>${apiDocsUrl}</a>",
-            401
-        );
-    }
-
-    /**
-     * @inheritdoc
-     */
-    public function supports(Request $request)
+    public function supports(Request $request): ?bool
     {
         if (!$request->headers->has('X-JWT-Authorization')) {
             return false;
         }
 
         $authorizationHeader = $request->headers->get('X-JWT-Authorization');
-        return preg_match('/^Token \S+$/', $authorizationHeader);
+        return (bool) preg_match('/^Token \S+$/', $authorizationHeader);
     }
 
-    /**
-     * @inheritdoc
-     */
-    public function getCredentials(Request $request)
+    public function authenticate(Request $request): PassportInterface
     {
         $authorizationHeader = $request->headers->get('X-JWT-Authorization');
         preg_match('/^Token (\S+)$/', $authorizationHeader, $matches);
-        return $matches[1];
-    }
-
-    /**
-     * @inheritdoc
-     *
-     * @param string $jwt the extracted JWT
-     */
-    public function getUser($jwt, UserProviderInterface $userProvider)
-    {
-        if (!$userProvider instanceof SessionUserProvider) {
-            throw new InvalidArgumentException(
-                sprintf(
-                    'The user provider must be an instance of SessionUserProvider (%s was given).',
-                    $userProvider::class
-                )
-            );
-        }
-
-        try {
-            $username = $this->jwtManager->getUserIdFromToken($jwt);
-        } catch (UnexpectedValueException $e) {
-            throw new CustomUserMessageAuthenticationException('Invalid JSON Web Token: ' . $e->getMessage());
-        } catch (Exception) {
-            throw new CustomUserMessageAuthenticationException('Invalid JSON Web Token');
-        }
-
-        /* @var SessionUserInterface $user */
-        $user = $userProvider->loadUserByUsername($username);
-
-
-        return $user;
-    }
 
-    /**
-     * @inheritdoc
-     *
-     * @param string $jwt the extracted JWT
-     * @param SessionUserInterface $user
-     *
-     * @return bool
-     */
-    public function checkCredentials($jwt, UserInterface $user)
-    {
-        if (!$user->isEnabled()) {
-            throw new CustomUserMessageAuthenticationException(
-                'Invalid JSON Web Token: user is disabled'
-            );
-        }
-
-        $tokenNotValidBefore = $user->tokenNotValidBefore();
-        $issuedAt = $this->jwtManager->getIssuedAtFromToken($jwt);
-        if ($tokenNotValidBefore) {
-            if ($tokenNotValidBefore > $issuedAt) {
-                throw new CustomUserMessageAuthenticationException(
-                    'Invalid JSON Web Token: Not issued after ' . $tokenNotValidBefore->format('c') .
-                    ' issued on ' . $issuedAt->format('c')
+        if (preg_match('/^Token (\S+)$/', $authorizationHeader, $matches)) {
```

Comment:
This seems to be a duplicate line. Given the check in `supports()` I'm not sure the `if` check is needed here. If it is needed the line above this can probably be removed.

Reply:
nice catch, will see if i can remove this...

Reply:
:heavy_check_mark: